packages/chaire-lib-backend/src/config/auth/directToken.config.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+import { PassportStatic } from 'passport';
+import BearerStrategy from 'passport-http-bearer';
+import { IAuthModel, IUserModel } from '../../services/auth/authModel';
+import config from '../server.config';
+
+// Direct token login strategy: the token will be used as the username. If it
+// exists in the database, the user will be logged in, otherwise, it will be
+// automatically created.
+
+const isTokenValid = (token: string) => {
+    const tokenFormat = config.auth?.directToken?.tokenFormat;
+    return tokenFormat === undefined ? true : token.match(tokenFormat);
```

Comment:
token match return an array or null, we should probably check explicitely for those, instead of returning the matching array. 

Also should we check that the regex is valid ??

Reply:
(I would love to have a log message saying that the configured regex is invalid. It could be elsewhere, in the app startup maybe. Or I guess eventually in a config validator, but would be good to have at least a TODO)